noir-projects/aztec-nr/aztec/src/macros/notes/mod.nr

```diff
@@ -405,10 +441,14 @@ comptime fn generate_setup_payload(
         .append(new_args_list)
         .push_back(quote { storage_slot: Field })
         .join(quote {,});
-    let new_generators =
-        new_generators_list.push_back(quote { aztec::generators::G_slot }).join(quote {,});
+    let new_generators = new_generators_list
+        .push_back(quote { aztec::generators::G_slot })
+        .push_back(quote { aztec::generators::G_len })
+        .join(quote {,});
+    let merged_fields_len = indexed_fixed_fields.len() + indexed_nullable_fields.len() + 1; // +1 for storage_slot
     let new_scalars = new_scalars_list
         .push_back(quote { std::hash::from_field_unsafe(storage_slot) })
+        .push_back(quote { std::hash::from_field_unsafe($merged_fields_len) })
```

Comment:
I wonder if the storage_slot and len should go at the beginning, so that they're always in a fixed location, regardless of the number of items in the note struct?